code/src/HC_SR04_c.cpp

```diff
@@ -0,0 +1,24 @@
+#include "HC_SR04_c.hpp"
+
+int16_t R2D2::Distance::HC_SR04_c::get_distance(){
+    trigger_pin.write(0);
+    hwlib::wait_us(2);
+    trigger_pin.write(1);
+    hwlib::wait_us(10);
+    trigger_pin.write(0);
+
+    while(!echo_pin.read()){
+        hwlib::wait_us(10);
```

Comment:
I am not sure if this wait serves a purpose: without rtos, the hwlib::wait functions are just while loops waiting for the given time to pass. I think the measurements would be more accurate without this function call.

Reply:
This wait is made for the future RTOS, nothing else